### PR TITLE
{Network} Fix `az network application-gateway private-link add` failed to create resources

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Description**

Fix bug https://github.com/Azure/azure-cli/issues/20708

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py [hunk 1]
```diff
@@ -1556,6 +1556,44 @@ def test_manage_appgw_private_endpoint(self, resource_group):
 
         self.cmd('network application-gateway private-link list -g {rg} --gateway-name {appgw} ')
 
+    @live_only()
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py:1559]:
```suggestion
```

**Review by kairu-ms — COMMENTED** [review]
'test_manage_appgw_private_endpoint.yaml' should not be updated, because you don't change that test case.

**Comment by yonzhan** [comment]
Network

**Review by wangzelin007 — APPROVED** [review]

**Comment by wangzelin007** [comment]
Please fix CI errors.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**